server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1212,26 +1213,42 @@ public synchronized void updateVisibleCheckpointForShard(final String allocation
      */
     public synchronized void setLatestReplicationCheckpoint(ReplicationCheckpoint checkpoint) {
         assert indexSettings.isSegRepEnabled();
-        assert handoffInProgress == false;
```

Comment:
Why this assertion is removed ?

Reply:
This doesn't fail but It felt unnecessary because we aren't handing off this context. As soon as the new primary shard is moved to primaryMode it will start its own replication timers.